packages/steiger/src/models/config/process-configuration/create-rule-instructions.test.ts

```diff
@@ -0,0 +1,113 @@
+import { describe, expect, it } from 'vitest'
+import createRuleInstructions from './create-rule-instructions'
+import { Config } from '@steiger/types'
+
+describe('createRuleInstructions', () => {
+  it('should create rule instructions for each rule present in the config', () => {
+    const config: Config = [
+      {
+        rules: {
+          rule1: 'warn',
+          rule2: 'error',
+        },
+        files: ['src/shared/ui/**/*', 'src/entities/user/ui/**/*'],
+        ignores: ['src/entities/user/ui/index.ts'],
+      },
+      {
+        rules: {
+          rule3: 'off',
+        },
+      },
+    ]
+
+    const result = createRuleInstructions(config)
+
+    expect(result).toEqual({
+      rule1: {
+        options: null,
+        globGroups: [
+          {
+            severity: 'warn',
+            files: ['src/shared/ui/**/*', 'src/entities/user/ui/**/*'],
+            ignores: ['src/entities/user/ui/index.ts'],
+          },
+        ],
+      },
+      rule2: {
+        options: null,
+        globGroups: [
+          {
+            severity: 'error',
+            files: ['src/shared/ui/**/*', 'src/entities/user/ui/**/*'],
+            ignores: ['src/entities/user/ui/index.ts'],
+          },
+        ],
+      },
+      rule3: {
+        options: null,
+        globGroups: [
+          {
+            severity: 'off',
+            files: [],
+            ignores: [],
+          },
+        ],
+      },
+    })
+  })
+
+  it('should add several glob groups for a rule if they are provided in the config', () => {
+    const config: Config = [
+      {
+        rules: {
+          rule1: 'warn',
+        },
+        files: ['src/shared/ui/**/*'],
+        ignores: ['src/shared/ui/index.ts'],
+      },
+      {
+        files: ['src/entities/user/ui/**/*'],
+        rules: {
+          rule1: 'error',
+        },
+      },
+    ]
+
+    expect(createRuleInstructions(config)).toEqual({
+      rule1: {
+        options: null,
+        globGroups: [
+          {
+            severity: 'warn',
+            files: ['src/shared/ui/**/*'],
+            ignores: ['src/shared/ui/index.ts'],
+          },
+          {
+            severity: 'error',
+            files: ['src/entities/user/ui/**/*'],
+            ignores: [],
+          },
+        ],
+      },
+    })
+  })
+
+  it('should throw an error if a rule has multiple options provided', () => {
+    const config: Config = [
+      {
+        rules: {
+          rule1: ['warn', {}],
+          rule2: ['error', {}],
+        },
+      },
+      {
+        files: ['src/shared/ui/**/*', 'src/entities/user/ui/**/*'],
+        rules: {
+          rule1: ['warn', {}],
```

Comment:
issue: I think we shouldn't throw errors when the specified options are the same. I believe that it's not immediately obvious what's the correct way of setting different severities while retaining the same options (even though we will guide users with error messages), and I anticipate that some people will want to simply duplicate the options object rather than omitting it, and since such a config doesn't contain any contradictions, we shouldn't throw errors at the user in this case.

I just think that running up against an error message is always frustrating for the user, so we should always try to minimize the error surface

suggestion: let's adjust this test to check for not throwing, then add another test that checks that an error will be thrown if two different options are specified

Reply:
100%, that's a good point. I'll make the fixes